core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -172,6 +172,13 @@ class Partition(val topic: String,
       // add replicas that are new
       allReplicas.foreach(replica => getOrCreateReplica(replica))
       val newInSyncReplicas = leaderAndIsr.isr.map(r => getOrCreateReplica(r)).toSet
+
+      // If the ISR size received from a LeaderAndIsrRequest is smaller than the existing ISR size,
+      // mark the isrShrinkRate meter. For example: This can happen if one of the followers is shutdown.
+      // Note that this metric is not triggered on follower -> leader transition
```

Comment:
Not sure what "follower -> leader transition" means. Do you mean this metric doesn't need to be updated in makeFollower()?